libs/knowledge-store/ragstack_knowledge_store/cassandra.py

```diff
@@ -362,65 +366,62 @@ def add_href_url_pairs(href_ids, url_ids):
 
         return ids
 
-    # TODO: Async
     @classmethod
     def from_texts(
-        cls,
-        texts: List[str],
+        cls: Type["CassandraKnowledgeStore"],
+        texts: Iterable[str],
         embedding: Embeddings,
         metadatas: Optional[List[dict]] = None,
+        ids: Optional[Iterable[str]] = None,
         **kwargs: Any,
-    ) -> "KnowledgeStore":
-        """Return VectorStore initialized from texts and embeddings."""
+    ) -> "CassandraKnowledgeStore":
+        """Return CassandraKnowledgeStore initialized from texts and embeddings."""
+        store = cls(embedding, **kwargs)
+        store.add_texts(texts, metadatas, ids=ids)
+        return store
 
-        ids = kwargs.pop("ids")
-        knowledge_store = KnowledgeStore(embedding, **kwargs)
-        knowledge_store.add_texts(texts, metadatas, ids=ids)
-        return knowledge_store
+    @classmethod
+    def from_documents(
+        cls: Type["CassandraKnowledgeStore"],
+        documents: Iterable[Document],
+        embedding: Embeddings,
+        ids: Optional[Iterable[str]] = None,
```

Comment:
Feels like this should take `metadatas` as well?

Reply:
Metadatas is a field of `Document`